orangecontrib/prototypes/widgets/owquickselect.py

```diff
@@ -0,0 +1,140 @@
+            None, self, "variable", contentsLength=50,
+            model=self.var_model, callback=self._on_variable_changed)
+
+        self.value_model = PyListModel()
+        value_combo = gui.comboBox(
+            None, self, "value", label="Value: ",
+            model=self.value_model, callback=self._on_value_changed,
+            contentsLength=50,
+            sizePolicy=(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
+
+        form.addRow("Variable:", var_combo)
+        form.addRow("Value:", value_combo)
+
+    @Inputs.data
+    def set_data(self, data):
+        self.closeContext()
+        self.data = data
+        self.Error.no_categorical.clear()
+        # TODO: Check that contexts are retrieved properly, also when removing
+        # and re-adding a connection
+
+        if data:
+            self.var_model.set_domain(data.domain)
+            self.info.set_input_summary(
+                len(data), format_summary_details(data))
+            if not self.var_model.rowCount():
+                self.data = None
+                self.Error.no_categorical()
+        else:
+            self.var_model.set_domain(None)
+            self.info.set_input_summary(self.info.NoInput)
+
+        self.variable = self.var_model[0] if self.data else None
+        self.openContext(self.data)
+        self.set_value_list()
+        if self.variable and self.pending_value in self.variable.values:
+            self.value = self.pending_value
+        self.commit()
+
+    def set_value_list(self):
+        if self.variable is None:
+            self.value_model.clear()
+            self.value = ""
+        else:
+            self.value_model[:] = self.variable.values
+            self.value = self.value_model[0] if self.variable.values else ""
+
+    def _on_variable_changed(self):
+        self.set_value_list()
+        self.commit()
+
+    def _on_value_changed(self):
+        self.pending_value = self.value
+        self.commit()
+
+    def commit(self):
+        if not (self.data and self.variable and self.value):
+            annotated = matching = unmatched = None
+            self.n_matched = None
+        else:
+            column = self.data.get_column_view(self.variable)[0]
+            valind = self.variable.values.index(self.value)
+            mask = column == valind
+            annotated = create_annotated_table(self.data, np.flatnonzero(mask))
+            matching = self.data[mask]
+            unmatched = self.data[~mask]
+            self.n_matched = len(matching)
+
+        self.Outputs.matching.send(matching)
+        self.Outputs.unmatched.send(unmatched)
+        self.Outputs.annotated.send(annotated)
+
+    def send_report(self):
+        if not self.data:
+            return
+
+        self.report_items(
+            "",
+            [("Filter", f"{self.variable.name} = '{self.value}'"),
+             ("Matching instances",
+              f"{self.n_matched} (out of {len(self.data) - self.n_matched})")]
```

Comment:
This is, in my opinion, incorrect.

It should say `f"{self.n_matched} (out of {len(self.data)})"`.

Reply:
"In my opinion" was entirely unnecessary. "Obviously" would be quite in place and well deserved.

I guess I initially planned to show the number of matching and unmatched.